### PR TITLE
Jennyf/logger issue in b2 cpm sample for msal with object null reference help

### PR DESCRIPTION
**Pull request by henrik-me**
Ensuring the PlatformInformation is initialized. Still a few more things to validate however sharing allowing additional feedback.

### CODE REVIEW

**Comment by henrik-me** [comment]
aside: the reference to the sample app should be removed.

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/ModuleInitializer.cs [hunk 1]
```diff
@@ -59,6 +59,9 @@ public static void EnsureModuleInitialized()
                 {
                     CoreLoggerBase.Default = new AdalLogger(Guid.Empty);
                     CoreTelemetryService.InitializeCoreTelemetryService(Telemetry.GetInstance() as ITelemetry);
+                    // Several statics in the library depends on platform information being timely initialized. 
+                    // The static initializer on PlatformInformationBase will ensure this gets done.
+                    new PlatformInformation();
```
Comment (jmprieur) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/ModuleInitializer.cs:64]: This looks like the right fix to me. I had not realized ADAL.NET was also affected.

**Review by jmprieur — APPROVED** [review]
LGTM. Elegant.
I closed #1232

File: core/src/MsalIdHelper.cs [hunk 2]
```diff
@@ -68,6 +68,10 @@ internal static class MsalIdHelper
     {
         public static IDictionary<string, string> GetMsalIdParameters()
         {
+            if (CorePlatformInformationBase.Instance == null)
+            {
+                throw CoreExceptionFactory.Instance.GetClientException("someErrorCode", "Platform Not Supported");
```
Comment (bgavrilMS) [inline core/src/MsalIdHelper.cs]: nit: would be helpful to put the platform here as well, platform information has it.

File: msal/tests/Test.NetCore21.Unit/Test.MSAL.NET.dotnetcore.Unit.csproj [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
```
Comment (bgavrilMS) [inline msal/tests/Test.NetCore21.Unit/Test.MSAL.NET.dotnetcore.Unit.csproj]: very cool!

**Review by bgavrilMS — APPROVED** [review]

File: core/src/MsalIdHelper.cs (hunk 2 above)
Comment (henrik-me) [reply core/src/MsalIdHelper.cs]: yeah, this whole method needs updating including the class calling it. So many re-initializations, wastes CPU at very little Memory cost. Considering whether I want to add that here or not. This part is in Core though, which doesn't have access to the full PlatformInformation class.

File: msal/tests/Test.NetCore21.Unit/Test.MSAL.NET.dotnetcore.Unit.csproj (hunk 3 above)
Comment (jennyf19) [reply msal/tests/Test.NetCore21.Unit/Test.MSAL.NET.dotnetcore.Unit.csproj]: I agree...this is great.

---
In reply to: [213117773](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1233#discussion_r213117773) [](ancestors = 213117773)

**Review by jennyf19 — APPROVED** [review]
I like the changes! 🚀

**Comment by henrik-me** [comment]
Merging to dev. Will have to work more on getting the .netcore tests to run on VSTS.

**Pull request merged**